fallback_test.go

```diff
@@ -52,9 +57,24 @@ func (c *testProxyCreator) New(trace plugin.Handler, upstream proxy.Upstream) pl
 		return nil
 	}
 
+	if c.handler != nil {
+		return c.handler
+	}
 	return &proxy.Proxy{Trace: trace, Upstreams: &[]proxy.Upstream{upstream}}
```

Comment:
@ekleiner , just a thought...

Maybe returning an instance of `proxy.Proxy` here is actually not very useful. How about always returning `c.handler` instead?

Looks good, btw!!

Reply:
.... and default to `dummyHandler`.